src/metrics/src/rusage.rs

```diff
@@ -5,55 +5,110 @@
+    type Gauge;
+    /// Libc type
+    type From;
+    /// Gauge type.
+    type To;
+    /// Convert an actual value.
+    fn from(value: Self::From) -> Self::To;
+}
+
+/// Unit for converting POSIX timeval.
+struct Timeval;
+impl Unit for Timeval {
+    type Gauge = Gauge;
+    type From = libc::timeval;
+    type To = f64;
+    fn from(Self::From { tv_sec, tv_usec }: Self::From) -> Self::To {
+        // timeval can capture negative values; it'd be surprising to see a negative values here.
+        (Duration::from_secs(u64::cast_from(tv_sec.abs_diff(0)))
+            + Duration::from_micros(u64::cast_from(tv_usec.abs_diff(0))))
+        .as_secs_f64()
+    }
+}
+
+/// Unit for direct conversion to i64.
+struct Unitless;
+impl Unit for Unitless {
+    type Gauge = IntGauge;
+    type From = libc::c_long;
+    type To = i64;
+    fn from(value: Self::From) -> Self::To {
+        value
+    }
+}
+
+/// Unit for converting KiB values to bytes expressed i64.
+struct KilobytesToBytes;
+impl Unit for KilobytesToBytes {
+    type Gauge = IntGauge;
+    type From = libc::c_long;
+    type To = i64;
+    fn from(value: Self::From) -> Self::To {
+        value.saturating_mul(1024)
+    }
+}
+
 metrics! {
     mz_metrics_libc
-    (ru_maxrss, "maximum resident set size"),
-    (ru_ixrss, "integral shared memory size"),
-    (ru_idrss, "integral unshared data size"),
-    (ru_isrss, "integral unshared stack size"),
-    (ru_minflt, "page reclaims (soft page faults)"),
-    (ru_majflt, "page faults (hard page faults)"),
-    (ru_nswap, "swaps"),
-    (ru_inblock, "block input operations"),
-    (ru_oublock, "block output operations"),
-    (ru_msgsnd, "IPC messages sent"),
-    (ru_msgrcv, "IPC messages received"),
-    (ru_nsignals, "signals received"),
-    (ru_nvcsw, "voluntary context switches"),
-    (ru_nivcsw, "involuntary context switches")
+    (ru_utime, "user CPU time used", "_seconds_total", Timeval),
+    (ru_stime, "system CPU time used", "_seconds_total", Timeval),
+    (ru_maxrss, "maximum resident set size", "_bytes", KilobytesToBytes),
+    (ru_ixrss, "integral shared memory size (unmaintained)", "", Unitless),
+    (ru_idrss, "integral unshared data size (unmaintained)", "", Unitless),
+    (ru_isrss, "integral unshared stack size (unmaintained)", "", Unitless),
+    (ru_minflt, "page reclaims (soft page faults)", "_total", Unitless),
+    (ru_majflt, "page faults (hard page faults)", "_total", Unitless),
+    (ru_nswap, "swaps (unmaintained)", "", Unitless),
+    (ru_inblock, "block input operations", "_total", Unitless),
+    (ru_oublock, "block output operations", "_total", Unitless),
+    (ru_msgsnd, "IPC messages sent (unmaintained)", "", Unitless),
+    (ru_msgrcv, "IPC messages received (unmaintained)", "", Unitless),
+    (ru_nsignals, "signals received (unmaintained)", "", Unitless),
+    (ru_nvcsw, "voluntary context switches", "_total", Unitless),
+    (ru_nivcsw, "involuntary context switches", "_total", Unitless)
```

Comment:
What do you think about removing the "unmaintained" metrics? I assume their values would always be zero anyway?

Reply:
Removed! They might be maintained on Mac, but since we're not running on anything but Linux in production there is little value in surfacing these metrics.